radix-engine/src/blueprints/consensus_manager/validator.rs

```diff
@@ -1607,20 +1630,18 @@ impl ValidatorBlueprint {
         total_stake_xrd_amount: Decimal,
         total_stake_unit_supply: Decimal,
     ) -> Result<Decimal, RuntimeError> {
-        let stake_unit_amount = if total_stake_xrd_amount.is_zero() {
-            xrd_amount
+        if total_stake_xrd_amount.is_zero() {
+            Ok(xrd_amount)
         } else {
             xrd_amount
-                .safe_div(total_stake_xrd_amount)
-                .and_then(|d| d.safe_mul(total_stake_unit_supply))
-                .ok_or_else(|| {
-                    RuntimeError::ApplicationError(ApplicationError::ValidatorError(
+                .safe_mul(total_stake_unit_supply)
```

Comment:
We should do division before multiplication here to reduce chance of overflow as well as be more consistent with pools